contracts/liquidityMining/KyberRewardLocker.sol

```diff
@@ -0,0 +1,317 @@
+  using SafeCast for uint256;
+
+  using SafeERC20 for IERC20Ext;
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  struct VestingSchedule {
+    uint64 startTime;
+    uint64 endTime;
+    uint128 quantity;
+  }
+
+  struct VestingSchedules {
+    uint256 length;
+    mapping(uint256 => VestingSchedule) data;
+  }
+
+  uint256 private constant MAX_REWARD_CONTRACTS_SIZE = 10;
+
+  /// @dev whitelist of reward contracts
+  mapping(IERC20Ext => EnumerableSet.AddressSet) internal rewardContractsPerToken;
+
+  /// @dev vesting schedule of an account
+  mapping(address => mapping(IERC20Ext => VestingSchedules)) private accountsVestingSchedules;
+
+  /// @dev An account's total escrowed balance per token to save recomputing this for fee extraction purposes
+  mapping(address => mapping(IERC20Ext => uint256)) public accountEscrowedBalance;
+
+  /// @dev An account's total vested reward per token
+  mapping(address => mapping(IERC20Ext => uint256)) public accountVestedBalance;
+
+  /// @dev where slashing tokens goes
+  mapping(IERC20Ext => address) public slashingTargets;
+
+  /// @dev lock time
+  mapping(IERC20Ext => uint256) public lockTime;
+
+  /* ========== EVENTS ========== */
+  event RewardContractAdded(address indexed rewardContract, bool isAdded);
+  event SetSlashingTarget(IERC20Ext indexed token, address target);
+  event SetLockTime(IERC20Ext indexed token, uint256 _lockTime);
+
+  /* ========== MODIFIERS ========== */
+
+  modifier onlyRewardsContract(IERC20Ext token) {
+    require(rewardContractsPerToken[token].contains(msg.sender), 'onlyRewardcontract');
+    _;
+  }
+
+  constructor(address _admin) PermissionAdmin(_admin) {}
+
+  /**
+   * @notice Add a whitelisted rewards contract
+   */
+  function addRewardsContract(IERC20Ext token, address _rewardContract) external onlyAdmin {
+    require(
+      rewardContractsPerToken[token].length() < MAX_REWARD_CONTRACTS_SIZE,
+      'rewardContracts is too long'
+    );
+    require(rewardContractsPerToken[token].add(_rewardContract), '_rewardContract is added');
+
+    emit RewardContractAdded(_rewardContract, true);
+  }
+
+  /**
+   * @notice Remove a whitelisted rewards contract
+   */
+  function removeRewardsContract(IERC20Ext token, address _rewardContract) external onlyAdmin {
+    require(rewardContractsPerToken[token].remove(_rewardContract), '_rewardContract is removed');
+
+    emit RewardContractAdded(_rewardContract, false);
+  }
+
+  function setSlashingTarget(IERC20Ext token, address target) external onlyAdmin {
+    slashingTargets[token] = target;
+
+    emit SetSlashingTarget(token, target);
+  }
+
+  function setLockTime(IERC20Ext token, uint256 _lockTime) external onlyAdmin {
+    lockTime[token] = _lockTime;
+
+    emit SetLockTime(token, _lockTime);
+  }
+
+  function lock(
+    IERC20Ext token,
+    address account,
+    uint256 quantity
+  ) external override {
+    lockWithStartTime(token, account, quantity, _blockTimestamp());
```

Comment:
Should be able to replace `_blockTimestamp()` with `block.timestamp` unless mocking it in tests

Reply:
Yeah but if we replace the code, it will not show these lines in the coverage.

Reply:
okay